File: ContainsDuplicate.java
import java.util.HashMap;
import java.util.Map;

/**
 * Created by yhf on 15/5/25.
 */

/*
Given an array of integers, find if the array contains any duplicates. Your function should return true if
any value appears at least twice in the array, and it should return false if every element is distinct.
*/

public class ContainsDuplicate {
    public static boolean containsDuplicate(int[] nums) {
        Map<Integer, Boolean> map = new HashMap<>();
        for (int num : nums) {
            if (map.containsKey(num)) {
                return true;
            } else {
                map.put(num, true);
            }
        }
        return false;
    }

    public static void main(String[] args) {

    }
}

File: Leetcode-Python/ContainsDuplicate.py
"""
Given an array of integers, find if the array contains any duplicates. Your function should return true if any value
appears at least twice in the array, and it should return false if every element is distinct.
"""


class Solution:
    # @param {integer[]} nums
    # @return {boolean}
    def containsDuplicate(self, nums):
        s = set()
        for num in nums:
            if s.__contains__(num):
                return True
            s.add(num)
        return False


if __name__ == '__main__':
    s = Solution()
    assert s.containsDuplicate([1, 2, 2]) == True
    assert s.containsDuplicate([1, 2, 3]) == False

